pynest/nest/lib/hl_api_simulation.py

```diff
@@ -382,11 +382,31 @@ def SetKernelStatus(params):
     GetKernelStatus
 
     """
-
+    for key in params.keys():
+        readonly = _sks_params.get(key)
+        if readonly is None:
+            # If the parameter is not in the docstring
+            raise KeyError(f'`{key}` is not a valid kernel parameter, '
+                           'valid parameters are: '
+                           ', '.join(f"'{p}'" for p in _sks_params.keys()))
+        elif readonly:
+            # If the parameter is tagged as read only
+            raise KeyError(f'`{key}` is a read only parameter and cannot '
+                           'be defined using SetKernelStatus')
     sps(params)
     sr('SetKernelStatus')
 
 
+# Parse the `SetKernelStatus` docstring to obtain all valid and readonly params
+lines = SetKernelStatus.__doc__.split('\n')
+# Get the lines describing parameters
+param_lines = (line for line in lines if ' : ' in line)
+# Excluding the first parameter `params` belonging to the function signature.
+next(param_lines)
+_sks_params = {ln.split(":")[0].strip(): "read only" in ln for ln in param_lines}
```

Comment:
Since the line has now been stripped above this can be changed to:

```python
_sks_params = {ln.split(" :")[0]: "read only" in ln for ln in param_lines}
```

Reply:
Done